pkg/asset/installconfig/aws/validation.go

```diff
@@ -330,15 +330,50 @@ var requiredServices = []string{
 // ValidateForProvisioning validates if the install config is valid for provisioning the cluster.
 func ValidateForProvisioning(session *session.Session, ic *types.InstallConfig, metadata *Metadata) error {
 	allErrs := field.ErrorList{}
-	allErrs = append(allErrs, validateExistingHostedZone(session, ic, metadata)...)
+	if ic.AWS.HostedZone == "" {
+		allErrs = append(allErrs, validatePreExistingPublicDNS(session, ic, metadata)...)
+	} else {
+		allErrs = append(allErrs, validateExistingHostedZone(session, ic, metadata)...)
+	}
 	return allErrs.ToAggregate()
 }
 
-func validateExistingHostedZone(session *session.Session, ic *types.InstallConfig, metadata *Metadata) field.ErrorList {
-	if ic.AWS.HostedZone == "" {
+// validate no pre-existing DNS records exist in the public DNS zone
+func validatePreExistingPublicDNS(session *session.Session, ic *types.InstallConfig, metadata *Metadata) field.ErrorList {
```

Comment:
Can we change the name of this function to indicate that it is validating that there is **not** a pre-existing public DNS record?
```suggestion
func validateNoExistingPublicDNS(session *session.Session, ic *types.InstallConfig, metadata *Metadata) field.ErrorList {
```